### PR TITLE
clientv3: only receive from closing streams in Watcher close

### PR DESCRIPTION
**Pull request by heyitsanthony**
Was overcounting the number of expected closing messages; the resuming list may have nil entries.

(test still throwing errors, so WIP)

### CODE REVIEW

File: clientv3/integration/watch_test.go [hunk 1]
```diff
@@ -868,3 +868,25 @@ func TestWatchCancelAndCloseClient(t *testing.T) {
...
+// then closes the watcher interface to ensure correct clean up.
+func TestWatchStressResumeClose(t *testing.T) {
+	defer testutil.AfterTest(t)
+	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 1})
+	defer clus.Terminate(t)
+	cli := clus.Client(0)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	wchs := make([]clientv3.WatchChan, 10000)
+	for i := range wchs {
```
Comment (xiang90) [inline clientv3/integration/watch_test.go]: good luck for the CPU of the CI.

File: clientv3/client.go [hunk 2]
```diff
@@ -86,6 +86,7 @@ func NewFromConfigFile(path string) (*Client, error) {
 // Close shuts down the client's etcd connections.
 func (c *Client) Close() error {
 	c.cancel()
+	c.Watcher.Close()
```
Comment (gyuho) [inline clientv3/client.go]: Wouldn't `c.conn.Close()` close the watcher as well? Why didn't we have this before?

**Comment by xiang90** [comment]
LGTM

File: clientv3/client.go (hunk 2 above)
Comment (xiang90) [inline clientv3/client.go]: same question

Comment (heyitsanthony) [inline clientv3/client.go]: Yes, but it's indirect. If there's a thousand running watches, the watch client shutdown path won't complete by the time of the goroutine leak check, so it looks like there are leaks. Directly closing the watch client ensures the client close will wait for the teardown to complete.

**Pull request merged**